includes/class-llms-forms.php

```diff
@@ -0,0 +1,885 @@
+	 * @return array|int|false `false` when the field isn't found in $fields, otherwise returns the field settings
+	 *                          as an array when `$return` is "field". Otherwise returns the field's index as an int.
+	 */
+	public function get_field_by( $fields, $key, $val, $return = 'field' ) {
+
+		foreach ( $fields as $index => $field ) {
+			if ( isset( $field[ $key ] ) && $val === $field[ $key ] ) {
+				return 'field' === $return ? $field : $index;
+			}
+		}
+
+		return false;
+
+	}
+
+	/**
+	 * Retrieve the rendered HTML for the form at a given location.
+	 *
+	 * @since [version]
+	 *
+	 * @param string $location Form location, one of: "checkout", "registration", or "account".
+	 * @param array  $args Additioal arguments passed to the short-circuit filter in `get_form_post()`.
+	 * @return string
+	 */
+	public function get_form_html( $location, $args = array() ) {
+
+		$blocks = $this->get_form_blocks( $location, $args );
+		if ( ! $blocks ) {
+			return '';
+		}
+
+		$html = '';
+		foreach ( $blocks as $block ) {
+			$html .= render_block( $block );
+		}
+
+		/**
+		 * Modify the parsed array of LifterLMS Form Fields.
+		 *
+		 * @since [version]
+		 *
+		 * @param string $html Form fields HTML.
+		 * @param string $location Form location, one of: "checkout", "registration", or "account".
+		 * @param array $args Additioal arguments passed to the short-circuit filter in `get_form_post()`.
+		 */
+		return apply_filters( 'llms_get_form_html', $html, $location, $args );
+
+	}
+
+	/**
+	 * Retrieve the WP Post for the form at a given location.
+	 *
+	 * @since [version]
+	 *
+	 * @param string $location Form location, one of: "checkout", "registration", or "account".
+	 * @param array  $args Additioal arguments passed to the short-circuit filter.
+	 * @return WP_Post|false
+	 */
+	public function get_form_post( $location, $args = array() ) {
+
+		/**
+		 * Skip core lookup of the form for the request location and return a custom form post.
+		 *
+		 * @since [version]
+		 *
+		 * @param null|WP_Post $post Return a WP_Post object to short-circuit default lookup query.
+		 * @param string $location Form location. Either "checkout", "registration", or "account".
+		 * @param array $args Additional custom arguments.
+		 */
+		$post = apply_filters( 'llms_get_form_post_pre_query', null, $location, $args );
+		if ( is_a( $post, 'WP_Post' ) ) {
+			return $post;
+		}
+
+		$query = new WP_Query(
+			array(
+				'post_type'      => $this->post_type,
+				'posts_per_page' => 1,
+				// Only show published forms to end users but allow admins to "preview" drafts.
+				'post_status'    => current_user_can( $this->capability ) ? array( 'publish', 'draft' ) : 'publish',
+				'meta_query'     => array(
+					'relation' => 'AND',
+					array(
+						'key'   => '_llms_form_location',
+						'value' => $location,
+					),
+					array(
+						'key'   => '_llms_form_is_core',
+						'value' => 'yes',
+					),
+				),
```

Comment:
this is one of those cases we can "speed up" the query, by adding no_found_rows=>true as we don't need any pagination information, right?!